Document standard library types (#5431)
nextflow-io/nextflow: docs/script.md

Copying a map with the `+` operator is a safer way to modify maps in Nextflow, specifically when passing maps through channels. This way, a new instance of the map will be created, and any references to the original map won't be affected.
:::

See {ref}`stdlib-types-map` for the set of available map operations.

(script-operators)=

## Operators

Operators are symbols that perform specific functions on one or more values, and generally make code easier to read. This section highlights some of the most commonly used operators.

:::{note}
Operators in this context are different from *channel operators*, which are specialized functions for working with channels. See {ref}`channel-page` for more information.
:::

The `==` and `!=` operators can be used to test whether any two values are equal (or not equal):

```nextflow
assert 2 + 2 == 4
assert [2, 2] != [4]
assert 'two plus two' != 'four'
```

:::{tip}

> The `assert` keyword simply tests a condition and raises an error if the condition is false. Every assert that you see on this page should succeed if you execute it.

Reviewer: ```suggestion
The `assert` keyword simply tests a condition and raises an error if the condition is false. Every assert that you see on this page will succeed if executed.
```

Author: I like that confidence 😅
